pyuavcan/transport/can/media/socketcan/_socketcan.py

```diff
@@ -129,6 +129,8 @@ async def send(self, frames: typing.Iterable[Envelope], monotonic_deadline: floa
                     self._loop.sock_sendall(self._sock, self._compile_native_frame(f.frame)),
                     timeout=monotonic_deadline - self._loop.time(),
                 )
+            except OSError as err:
+                raise pyuavcan.transport.InvalidMediaConfigurationError("Invalid socketcan configuration") from err
```

Comment:
OSError can mean a lot of things. I would add an explicit check here like (requires `import errno`):

    if err.errno == errno.EINVAL and self._is_fd:
        ...

...and raise the MediaConfigurationError only if the check holds (otherwise just re-`raise` it as-is). Also, maybe the text should go like `Invalid socketcan configuration: the device probably does not support CAN FD; try Classic CAN` or something similar, the point is to add more information to aid the user in solving this problem.